test/slipstream/socket_test_test.exs

```diff
@@ -0,0 +1,193 @@
+defmodule Slipstream.SocketTestTest do
+  # Slipstream.SocketTest is an exunit case template
+  use Slipstream.SocketTest
+
+  @moduledoc """
+  A test suite that tests that a test suite tests tests
+
+  Note that this is not necessarily a good example to follow for testing
+  clients: in order to make assertions about the status of the client, we have
+  written a client fixture (`@fixture`) that sends messages to the test process
+  when meaningful state changes occur (on connection, e.g.). This is not
+  recommended for writing clients in the wild.
+  """
+  @moduledoc since: "0.2.0"
+
+  @fixture Slipstream.TestModeClient
+
+  setup do
+    [topic: "rooms:lobby", event: "msg:new", params: %{"fizz" => "buzz"}]
+  end
+
+  describe "given that the fixture is spawned" do
+    setup do
+      start_supervised!({@fixture, self()})
+
+      :ok
+    end
+
+    test "the client connects when we do an accept_connect/1" do
+      accept_connect(@fixture)
+
+      assert_receive {@fixture, :connected}
+    end
+
+    test "passing a nil client or fake name raises ArgumentError" do
+      assert_raise ArgumentError, ~r"cannot find pid for client nil", fn ->
+        accept_connect(nil)
+      end
+
+      assert_raise ArgumentError, ~r"cannot find pid for client :foo", fn ->
+        accept_connect(:foo)
+      end
+    end
+
+    test "passing a dead client will raise ArgumentError" do
+      pid = spawn(fn -> exit(:normal) end)
+      ref = Process.monitor(pid)
+
+      assert_receive {:DOWN, ^ref, :process, ^pid, :normal}
+      refute Process.alive?(pid)
+
+      assert_raise ArgumentError,
+                   ~r"cannot send to client #PID<[\d\.]+> that is not alive",
+                   fn ->
+                     accept_connect(pid)
+                   end
+    end
+
+    test "we may connect and join a topic all at once with connect_and_join/5",
+         c do
+      topic = c.topic
+      params = c.params
+
+      join(topic, params)
+
+      connect_and_join @fixture, ^topic, ^params, {:ok, params}
```

Comment:
why `join` AND `connect_and_join`?

Reply:
understanding more after looking more deeply at `assert_join`

should this name be `assert_connect_and_join` or `connect_and_assert_join`?

Reply:
I don't mind that second one

the `join/2` in this case is not from `Slipstream.SocketTest` but from `defp join/2` below

this isn't what it'll look in an actual client test. We have to tell the client to join here because it's just a crappy fixture we want to control imperatively from the tests (in this case with `Kernel.send(fixture, {:join, topic, params})`

`connect_and_join/5` is more meant for clients that attempt to join immediately after connecting

Reply:
renamed to `connect_and_assert_join/5`